ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlInternalTopicUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.exception.KafkaTopicExistsException;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
+import org.apache.kafka.common.config.TopicConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * KsqlInternalTopicUtils provides some utility functions for working with internal topics.
+ * An internal topic is a topic that the KSQL server uses to manage its own internal state.
+ * This is separate from KSQL sink topics, and topics internal to individual queries
+ * (e.g. repartition and changelog topics). Some examples are the command topic used by
+ * interactive KSQL, and the config topic used by headless KSQL.
+ */
+public class KsqlInternalTopicUtils {
+  private static final Logger log = LoggerFactory.getLogger(KsqlInternalTopicUtils.class);
+
+  private static final int NPARTITIONS = 1;
+
+  /**
+   * Compute a name for an internal topic.
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @param topicSuffix A suffix that is appended to the topic name.
+   * @return The computed topic name.
+   */
+  public static String getTopicName(final KsqlConfig ksqlConfig, final String topicSuffix) {
+    return String.format(
+        "%s%s_%s",
+        KsqlConstants.KSQL_INTERNAL_TOPIC_PREFIX,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+        topicSuffix
+    );
+  }
+
+  /**
+   * Ensure that an internal topic exists with the requested configuration, creating it
```

Comment:
docs is a bit misleading:

> Ensure that an internal topic exists with the requested configuration

If the topic already exists it doesn't check replia / partition count match whats being requested.

Reply:
It does now (no more catch(KafkaTopicExistsException)). There's a test to that effect as well.

Reply:
What am I missing then? This code doesn't look to check rf & partition count of an existing topic.

```java
// This if statement will return true if the topic exists...
if (topicClient.isTopicExists(name)) {
      final ImmutableMap<String, Object> requiredConfig =
          ImmutableMap.of(TopicConfig.RETENTION_MS_CONFIG, requiredTopicRetention);

       // This call only updates retention...
       if (topicClient.addTopicConfig(name, requiredConfig)) {
        log.info(
            "Corrected retention.ms on ksql internal topic. topic:{}, retention.ms:{}",
            name,
            requiredTopicRetention);
      }

       // And then the function returns...
       return;
    }

// So where does it check replica and partition count?
```

Reply:
ah good call. I just got rid of the catch at the end (the exception is only thrown if the replica/partition-counts differ